Submitted By: Ryan Oliver (ryan dot oliver at pha dot com dot au)
Origin: Ryan Oliver
Date: 2005-03-21
Initial Package Version: 3.4.3 
Description: This patch is primarily here to fix issues with libtool when building
	packages on a multi-arch/multilib system, as libtool effectively uses the
	library search paths from gcc -print-search-dirs to locate .la files.
	This poses a problem as by default gcc supplies
		eg: /usr/lib64/gcc/x86_64-pc-linux-gnu/3.4.3/../../../
	which following the example would cause libtool to use .la files under
	/usr/lib64 instead of /usr/lib if building a package 32bit, and thus will
	attempt to link in incompatible 64bit libraries.
	Here we simply remove this path ( provided by standard_startfile_prefix )
	from the library search order

diff -uNr gcc-3.4.3-orig/gcc/gcc.c gcc-3.4.3/gcc/gcc.c
--- gcc-3.4.3-orig/gcc/gcc.c	2004-09-09 01:16:11.000000000 +1000
+++ gcc-3.4.3/gcc/gcc.c	2005-03-21 17:54:44.000000000 +1100
@@ -6135,7 +6135,7 @@
 	add_sysrooted_prefix (&startfile_prefixes,
 			      standard_startfile_prefix, "BINUTILS",
 			      PREFIX_PRIORITY_LAST, 0, NULL, 1);
-      else if (*cross_compile == '0')
+      /* else if (*cross_compile == '0')
 	{
 	  if (gcc_exec_prefix)
 	    add_prefix (&startfile_prefixes,
@@ -6147,7 +6147,7 @@
 			      machine_suffix,
 			      standard_startfile_prefix, NULL),
 		      NULL, PREFIX_PRIORITY_LAST, 0, NULL, 1);
-	}
+	} */
 
       add_sysrooted_prefix (&startfile_prefixes, standard_startfile_prefix_1,
 			    "BINUTILS", PREFIX_PRIORITY_LAST, 0, NULL, 1);
